src/Features/Core/Diagnostics/EngineV1/DiagnosticAnalyzerDriver.cs

```diff
@@ -279,9 +281,10 @@ private ImmutableArray<SyntaxNode> GetSyntaxNodesToAnalyze()
 
                 if (diagnostics.Count == 0)
                 {
-                    return SpecializedCollections.EmptyEnumerable<Diagnostic>();
+                    return ImmutableArray<Diagnostic>.Empty;
                 }
 
+                WorkspaceAnalyzerManager.UnregisterAnalyzerExceptionDiagnosticHandler(handler);
```

Comment:
To basically get the project info, otherwise diagnostic shows with no project name in error list, even though we know which project's analysis caused an exception. And this information is very helpful especially for per-project analyzers.